### PR TITLE
WIP news: initial commit

### PR DESCRIPTION
**Pull request by kasioumis**
* Initial port of the Webnews module from legacy
  to the news module in master.

Co-authored-by: Nikolaos Kasioumis <nikolaos.kasioumis@cern.ch>
Signed-off-by: Robert Winchell <robertzwinchell@gmail.com>

### CODE REVIEW

File: invenio/modules/news/bundles.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Search bundles."""
+
+from invenio.ext.assets import Bundle
```
Comment (hachreak) [inline invenio/modules/news/bundles.py]: can you add here and in config.py `from __future__ import unicode_literals`? thanks :)

File: invenio/modules/news/models.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    body = db.Column(db.String(512), nullable=False, server_default='0')
+    target_element = db.Column(
+        db.String(256),
+        nullable=False,
+        server_default='0'
+    )
+    target_page = db.Column(db.String(256), nullable=False)
+
+    @property
+    def serialize(self):
```
Comment (jirikuncar) [inline invenio/modules/news/models.py]: You should be able to use `to_dict()` or just `dict(self)`.

File: invenio/modules/news/static/css/custom.css [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*
```
Comment (jirikuncar) [inline invenio/modules/news/static/css/custom.css]: this file should be called: `invenio/modules/news/static/css/news/base.css

File: invenio/modules/news/templates/admin/Edit.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{#
```
Comment (jirikuncar) [inline invenio/modules/news/templates/admin/Edit.html]: - [ ] all files should be lowercased
- [ ] rename to `invenio/modules/news/templates/news/edit.html`

File: invenio/modules/news/templates/admin/Edit_base.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
+{#
```
Comment (jirikuncar) [inline invenio/modules/news/templates/admin/Edit_base.html]: - [ ] rename to invenio/modules/news/templates/news/edit_base.html

File: invenio/modules/news/templates/admin/Edit_base.html [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+{% block page_body %}
+
+
+
+{% block body %}
+{{ super() }}
+
+
+<div id="page-wrapper">
+	<div class="container-fluid">
```
Comment (jirikuncar) [inline invenio/modules/news/templates/admin/Edit_base.html]: use 2 spaces indentation

File: invenio/modules/news/templates/details_base.html [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                                </hr>
+                            </div>
+                            <div class="record-abstract">
+                                <small>
+                                    {{searchResult.body}}
+                                </small>
+                                <br>
+                                <br>
+
+                                <strong>Author: {{searchResult.nickname}}</strong><br>
```
Comment (jirikuncar) [inline invenio/modules/news/templates/details_base.html]: use translatable strings everywhere `_('Author')`

File: invenio/modules/news/views/admin.py [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+                )   # [news,tooltip,tag]
+        except IntegrityError:
+                flash('Error')
+                # alert = config.CFG_WEBNEWS_ERROR_ALERT
+
+
+# For Object Update
+@blueprint.route(config.CFG_WEBNEWS_ADMIN_UPDATE, methods=['GET', 'POST'])
+@login_required
+@permission_required(config.CFG_WEBNEWS_WEBACCESSACTION)
```
Comment (jirikuncar) [inline invenio/modules/news/views/admin.py]: create new action in `acl.py` and use it here

File: invenio/modules/news/views/webnews.py [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+    request,
+    session,
+)
+from flask.ext.menu import register_menu
+
+from invenio.ext.sqlalchemy import db
+
+from sqlalchemy.exc import IntegrityError
+
+from .. import config
```
Comment (jirikuncar) [inline invenio/modules/news/views/webnews.py]: never import config directly - use application config `current_app.config` or `invenio.base.globals:cfg`

File: invenio/modules/news/views/webnews.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    'webnews',
+    __name__,
+    template_folder='../templates',
+    static_folder='../static'
+)
+# @register_menu(blueprint, 'main.webnews',config.CFG_WEBNEWS_ADMIN_MAIN_NAV)
+# @register_menu(blueprint, 'webnews',config.CFG_WEBNEWS_ADMIN_MAIN_NAV)
+
+
+@blueprint.route(config.CFG_WEBNEWS_MENU_INDEX)
```
Comment (jirikuncar) [inline invenio/modules/news/views/webnews.py]: Routes should not be directly configurable, but you should have Blueprint prefix.

File: invenio/modules/news/views/webnews.py [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+                ) | NwsSTORY.body.contains(request.form['keywords'])
+            ).filter_by(document_status='SHOW').all()
+            return render_template(
+                'search.html',
+                searchResult=result,
+                resultshow='block',
+                EncodeStr=Encode
+            )
+        except IntegrityError:
+            flash('Error')
```
Comment (jirikuncar) [inline invenio/modules/news/views/webnews.py]: use correct flash message type and wrap the descriptive message into translatable string ...

File: invenio/modules/news/encoder.py [hunk 12]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
```
Comment (jirikuncar) [inline invenio/modules/news/encoder.py]: Use standard encryption for database columns from SQLAlchemy-Utils (see e.g. oauthserver.models). If you need to generate tokens then have a look to http://pythonhosted.org/itsdangerous/#url-safe-serialization

**Comment by jirikuncar** [comment]
@kasioumis please move it to new `invenio-news` repo. Thanks

**Comment by jirikuncar** [comment]
moved to https://github.com/inveniosoftware/invenio-news/pull/1

**Pull request closed without merging**